File: challenges/medium/01_student_grades/dispatcher.py
def build_handlers(event_types):
    """
    Given a list of event type names, return a dict mapping
    each event type to its handler function.

    Each handler accepts a payload and returns:
        {"event": <event_type>, "data": <payload>}
    """
    handlers = {}
    for event in event_types:
        handlers[event] = lambda payload: {"event": event, "data": payload}
    return handlers

File: challenges/medium/13_student_grades_ts/dispatcher.ts
export function buildHandlers(
  eventTypes: string[]
): Record<string, (payload: any) => { event: string; data: any }> {
  const handlers: Record<string, (payload: any) => any> = {};
  for (var event of eventTypes) {    // bug: var, not let
    handlers[event] = (payload: any) => ({ event: event, data: payload });
  }
  return handlers;
}
